### PR TITLE
[community] Pebblo: Policy-based semantic context in PebbloRetrievalQA

### PR DESCRIPTION
**Pull request by Raj725**
- **Description:** Policy-based semantic enforcement in PebbloRetrievalQA. With this update, they can configure policy-based semantic enforcement via the Pebblo cloud dashboard.
- **Issue:** NA
- **Dependencies:** NA
- **Tests:** Added

### CODE REVIEW

**Comment by Raj725** [comment]
@eyurtsev Please review

**Review by srics — APPROVED** [review]
LGTM

**Comment by Raj725** [comment]
@eyurtsev @ccurme Please review

**Review by srics — APPROVED** [review]

File: libs/community/langchain_community/chains/pebblo_retrieval/utilities.py [hunk 1]
```diff
@@ -334,6 +354,150 @@ async def acheck_prompt_validity(
...
+            if guardrail := self.policy_cache.user_semantic_guardrail.get(identity):
+                _all_guardrails.append(guardrail)
+
+        semantic_context = self._combine_all_semantic_filters(_all_guardrails)
+        return semantic_context
+
+    def _start_policy_refresh_thread(self) -> None:
+        """Start a thread to fetch policy from the Pebblo cloud."""
+        logger.info("Starting policy refresh thread.")
+        policy_thread = threading.Thread(target=self._fetch_policy, daemon=True)
```
Comment (efriis) [inline libs/community/langchain_community/chains/pebblo_retrieval/utilities.py:419]: I don't think starting background threads is desired - why would this be necessary

**Review by efriis — COMMENTED** [review]
one question, and if you're interested in maintaining this integration without us in the loop, we'd love to get an integration package out! Future PRs against langchain would just be {docs updates, as well as registering your package in `libs/packages.yml`, deprecating this community integration in favor of your integration package}

Here's the guide, and if you have questions, feel free to leave them in the comments on those pages so others can see them! https://python.langchain.com/docs/contributing/how_to/integrations/

**Comment by Raj725** [comment]
> one question, and if you're interested in maintaining this integration without us in the loop, we'd love to get an integration package out! Future PRs against langchain would just be {docs updates, as well as registering your package in `libs/packages.yml`, deprecating this community integration in favor of your integration package}
> 
> Here's the guide, and if you have questions, feel free to leave them in the comments on those pages so others can see them! https://python.langchain.com/docs/contributing/how_to/integrations/

@efriis Thank you for the suggestion! Just to confirm, when you mention the integration package, do you mean creating it as a separate package under `langchain-ai/langchain-pebblo`, or should it be under our own GitHub account, like `daxa-ai/langchain-pebblo`?

File: libs/community/langchain_community/chains/pebblo_retrieval/utilities.py (hunk 1 above)
Comment (Raj725) [reply libs/community/langchain_community/chains/pebblo_retrieval/utilities.py:419]: We will consider moving away from using threads and instead implement a cache with TTL

**Comment by efriis** [comment]
the latter! daxa-ai/langchain-pebblo

if that works for you, feel free to close this PR and do the work there! If you think you can replace the threads with a TTL implementation here in the next 5-7 days, feel free to keep this open and push that in, but if you're at all considering the external package would recommend that!

**Comment by efriis** [comment]
will close and assume external package is the way forward based on conversation!

**Pull request closed without merging**